### ISSUE TITLE
Get the statusCode from an UploadTask response

### ISSUE THREAD

**Issue by martyfuhry** [781flyingdutchman/background_downloader#255]
**Is your feature request related to a problem? Please describe.**
I would like to be able to get the `statusCode` of the server response from an upload task.

**Describe the solution you'd like**
The `TaskStatusUpdate` contains the `responseBody`, `responseHeaders` and `mimeType`, but I don't know how to access the `statusCode`. I suppose exposing it as a member variable here would be best.

**Describe alternatives you've considered**
I'm not tremendously sure how else to retrieve the status code from the response without access to it. I don't believe it is in the headers or the response body.

**Additional context**
I apologize if this is already supported, but I couldn't find any mention of accessing the `statusCode` for anything except for `Response` style objects.

Thank you!

**Comment by 781flyingdutchman** [comment]
Correct, a 200 or 201 won't generate an exception. I think I can add this, I'm just not in a position to implement it right now, so it will take about a week to get this out.

**Comment by martyfuhry** [comment]
Not a problem at all! Please let me know if there's anything I can do to help. I'm happy to make a PR if you'd like.

**Comment by 781flyingdutchman** [comment]
I do appreciate that! I think it's relatively easy to add (similar to how the responseheaders are done), but if you do it has to be done for all three platforms and all download and upload variants. If you're ok with that then please go ahead, otherwise I am obviously more familiar with the code base and happy to add it in a few days. 

Let me know, so we don't duplicate efforts.

**Comment by 781flyingdutchman** [comment]
Hi, I'm working on this (just need some testing) so please hold off on a PR 😉

**Comment by 781flyingdutchman** [comment]
Hi, any chance you can test this? Change your pubspec.yaml to:
```
background_downloader:
    git:
      url: https://github.com/781flyingdutchman/background_downloader.git
      ref: dev
```
The `TaskStatusUpdate` for tasks that complete successfully (or with .notFound) now include the `responseStatusCode` as a field.

Let me know if this works as you'd expect.

**Comment by 781flyingdutchman** [comment]
Implemented in V8.3.0
